enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,165 @@
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-02-21
+last-updated: 2020-02-21
+status: implementable
+---
+
+# baremetal-keepalived-unicast
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Baremetal platform uses VIPs (Virtual IP) and [Keepalived](https://www.keepalived.org/) to 
+support high availability for API and default Ingress.
+A VIP is used to provide failover of the service across the relevant nodes (including the bootstrap instance).
+
+Two VIPs are supported:
+* api-vip
+* ingress-vip
+
+The `api-vip` first resides on the bootstrap instance and it moves to one of the control plane nodes, but only after the bootstrap process has completed and the bootstrap instance is stopped. This ensures that the API on the control plane nodes is fully functional before the api-vip moves.
+The `ingress-vip` resides on one of the nodes that runs a router pod.
+
+In current implementation, Keepalived is configured to run [VRRP management protocol](https://tools.ietf.org/html/rfc3768#section-3) of the VIP addresses in a multicast mode, but since not all environments have multicast available, this proposal seeks to remedy that.
+
+
+## Motivation
+
+As mentioned above, currently Keepalived sends the VRRP management protocol via multicast. This requires a
+minimal amount of configuration, as the multicast peers do not need to be specified.
+However, there are likely to be environments where multicast has been disabled because
+of network policy. 
+
+Additionally, moving to unicast will also solve the possible collisions when two or more clusters are on the same broadcast/multicast domain. this use case described in more detail in `user story #2` below.
+
+In these cases, it is necessary to utilized unicast addresses within keepalived rather than make this an optional configuration. It makes sense to make it the default since unicast will "just work" in all environments.
+
+### Goals
+
+* A deployment is possible without the need for multicast on a network.
+* No loss of functionality from the current multicast implementation.
+
+### Non-Goals
+
+* A switchable implementation. It does not need to be an installer choice.
+* The control plane hosts on different subnets (not on the same L2 segment).
+
+## Proposal
+
+To enable keepalived to communicate via unicast instead of multicast, and to handle
+the fact that the bootstrap IP is unknown to the masters.
+
+### User Stories
+
+#### Story 1 - Allow deployment where networking doesn't permit multicast
+
+As an administrator, I want to deploy OpenShift on bare metal nodes with 
+installer-provisioned infrastructure (IPI) where the networking doesn't permit multicast.
```

Comment:
AFAICT this applies to all on-prem platforms regardless of IPI/UPI.

Reply:
Do we want to adopt unicast change also to other on-prem platforms? 
I thought that there is no such requirement.

Reply:
> AFAICT this applies to all on-prem platforms regardless of IPI/UPI.

I think UPI already implies that the users might be using their own networking setup not prescribed by OpenShift, like None platform or vSphere upi etc. So I think calling out baremetal-ipi (or others like that) seems better to me..

Reply:
agree that clarifying if this will be bare metal ipi only, or also openstack IPI and vsphere IPI, would be good

Reply:
I checked this issue with the PMs and seems that unicast requirement is relevant only for the baremetal-ipi case.

I updated the summary and goals sections.

Reply:
done